src/cpp/session/modules/NotebookData.R

```diff
@@ -52,7 +52,7 @@
 
     max.print <- if (is.null(options$max.print)) getOption("max.print", 1000) else options$max.print
 
-    x <- head(x, max.print)
+    x <- as.data.frame(head(x, max.print))
```

Comment:
@jmcphers I noticed that calling `is.data.frame()` on a `data.table` is `TRUE` so we need an extra param to `.rs.toDataFrame` to force this case.